workflows/src/main/kotlin/com/r3/corda/lib/tokens/workflows/utilities/TokenBuilder.kt

```diff
@@ -0,0 +1,84 @@
+package com.r3.corda.lib.tokens.workflows.utilities
+
+import com.r3.corda.lib.tokens.contracts.states.FungibleToken
+import com.r3.corda.lib.tokens.contracts.types.IssuedTokenType
+import com.r3.corda.lib.tokens.contracts.types.TokenType
+import com.r3.corda.lib.tokens.contracts.utilities.heldBy
+import com.r3.corda.lib.tokens.contracts.utilities.issuedBy
+import com.r3.corda.lib.tokens.contracts.utilities.of
+import net.corda.core.contracts.Amount
+import net.corda.core.identity.Party
+import java.lang.IllegalArgumentException
+import java.math.BigDecimal
+
+/**
+ * This is a utility class designed for Java developers to more easily
+ * access Kotlin infix functions to build token types.
+ */
+class TokenBuilder {
+    private var intAmount: Int? = null
```

Comment:
Why do you need 4 different amounts here? Can you convert everything to long at the point of assignment?

Reply:
This is an interesting point, I was trying to keep this class as lightweight as possible. The conversion to `Long` is done in a very specific way in `AmountUtilities.kt` was hoping to leverage the functionality there so as to not duplicate.